src/app/modules/verification/__tests__/verification.routes.spec.ts

```diff
@@ -0,0 +1,360 @@
+import bcrypt from 'bcrypt'
+import { getReasonPhrase, StatusCodes } from 'http-status-codes'
+import mongoose from 'mongoose'
+import session, { Session } from 'supertest-session'
+
+import { BasicField, IVerificationSchema } from 'src/types'
+
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import { buildCelebrateError } from 'tests/unit/backend/helpers/celebrate'
+import { generateDefaultField } from 'tests/unit/backend/helpers/generate-form-data'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import getVerificationModel from '../verification.model'
+import { VfnRouter } from '../verification.routes'
+
+import {
+  generateFieldParams,
+  MOCK_HASHED_OTP,
+  MOCK_OTP,
+  MOCK_RECIPIENT,
+  MOCK_SIGNED_DATA,
+} from './verification.test.helpers'
+
+const verificationApp = setupApp('/transaction', VfnRouter)
+const VerificationModel = getVerificationModel(mongoose)
+
+jest.mock('nodemailer', () => ({
+  createTransport: jest.fn().mockReturnValue({
+    sendMail: jest.fn().mockResolvedValue(true),
+  }),
+}))
+// Default Twilio export is a function
+jest.mock('twilio', () => () => ({
+  messages: {
+    create: jest.fn().mockResolvedValue({
+      sid: 'mockSid',
+    }),
+  },
+}))
+
+describe('verification.routes', () => {
+  let mockTransaction: IVerificationSchema
+  let mockTransactionId: string
+  let mockEmptyFormId: string
+  let mockVerifiableFormId: string
+  let mockEmailFieldId: string
+  let mockMobileFieldId: string
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+
+  beforeEach(async () => {
+    request = session(verificationApp)
+    jest.clearAllMocks()
+    await dbHandler.clearDatabase()
+
+    // Form without verifiable fields
+    const { form: emptyForm } = await dbHandler.insertEmailForm()
+    mockEmptyFormId = String(emptyForm._id)
+
+    // Form with verifiable fields
+    const emailField = generateDefaultField(BasicField.Email, {
+      isVerifiable: true,
+    })
+    const mobileField = generateDefaultField(BasicField.Mobile, {
+      isVerifiable: true,
+    })
+    mockEmailFieldId = String(emailField._id)
+    mockMobileFieldId = String(mobileField._id)
+    const { form: verifiableForm } = await dbHandler.insertEmailForm({
+      // Alternative mail domain so as not to clash with emptyForm
+      mailDomain: 'test2.gov.sg',
+      formOptions: {
+        form_fields: [emailField, mobileField],
+      },
+    })
+    mockVerifiableFormId = String(verifiableForm._id)
+
+    // Mock transaction with both email and mobile fields
+    mockTransaction = await VerificationModel.create({
+      formId: mockVerifiableFormId,
+      fields: [
+        generateFieldParams({
+          _id: mockEmailFieldId,
+          // Hash created 1 minute ago, so we can test requesting for new OTP
+          // without being rejected due to minimum waiting time
+          hashCreatedAt: new Date(Date.now() - 60 * 1000),
```

Comment:
should we consider adding `luxon` so that this is clearer? because as it is, 60 * 1000 is abit cryptic and requires prior knowledge that date arithmetic is in milliseconds